addons/base_vat/models/res_partner.py

```diff
@@ -222,7 +222,8 @@ def _build_vat_error_message(self, country_code, wrong_vat, record_label):
         else:
             company = self.env.company
 
-        vat_label = company.country_id.vat_label if company.country_id else "Tax ID"
+        country = self.env['res.country'].search([('code', '=ilike', country_code)], limit=1)
+        vat_label = country.vat_label or _("VAT")
```

Comment:
I am not convinced of this change, do you have a task ID I can refer?

For an Indian company with a Canadian customer:

Old:
The VAT number xxx for Partner yyy

Before:
The GSTIN number xxx for Partner yyy does not seem valid

Now:
The HST number xxx for Partner yyy ...
how does the user know what HST number is?
To me the original was the best

`_("VAT")` would require a translation in stable. 

In master, the `country_code` param should change to `country` - looks like a wasted `search` considering the caller always reads the country to retrieve the code.
@jco-odoo

Reply:
now it's

if the company is not Indian then
The VAT(GSTIN) number xxx for Partner yyy does not seem valid

if the company is Indian then
The GSTIN number xxx for Partner yyy does not seem valid

Reply:
I was convinced you would do the change like @h4818 says, but somehow I only realized now it is not like that.   If I am in Canada, I might not even be aware that the VAT number is called GSTIN.